content/articles/object-detection-with-yolov5-and-pytorch/index.md

```diff
@@ -0,0 +1,197 @@
+python detect.py --source 'https://youtu.be/Zgi9g1ksQHc'  # for a specific YouTube video
+python detect.py --source 'rtsp://example.com/media.mp4'  # for a specific RTSP, RTMP, or HTTP stream
+```
+
+>**N/B**: Before running the commands above, make sure the objects you want to detect are located at `data/...` e.g. `data/videos`, `data/images` etc.
+
+### Model inference with PyTorch Hub and YOLOv5
+
+To check the inference using PyTorch, we will load the pre-trained YOLOv5s model from PyTorch Hub and then pass an image for inference. 
+
+>N|B: `'YOLOv5s'` is the fastest and lightest YOLOv5 model.
+
+```python
+import torch
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
+
+# Image
+img = 'https://ultralytics.com/images/zidane.jpg'
+
+# Inference
+results = model(img)
+
+results.pandas().xyxy[0]
+```
+
+To see the results of the code above, feel free to access it on [GitHub.](https://github.com/FREDERICO23/object-detection)
+
+### Model training 
+
+Since the COCO dataset is automatically downloaded from [YOLOv5 latest releases](https://github.com/ultralytics/yolov5/releases), we will proceed to train our model.
+
+We will use YOLOv5 pretrained weights (`--weights yolov5s.pt`) or randomly initialized weights (`--weights '' --cfg yolov5s.yaml`)  when training our model as shown below:
+```bash
+python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt --cache
+```
+or
+```bash
+python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights '' --cfg yolov5s.yaml --cache
+```
+Our **training results** will be saved to the directory `runs/train/exp` with incrementing run directories, i.e. `runs/train/exp1`, `runs/train/exp2`...
+
+### Model validation
+
+To evaluate our model's accuracy on the [COCO](https://cocodataset.org/#home) dataset, we will download the [COCO 2017](https://github.com/ultralytics/yolov5/blob/74b34872fdf41941cddcf243951cdb090fbac17b/data/coco.yaml#L14) validation dataset using the following command:
+
+```bash
+torch.hub.download_url_to_file('https://ultralytics.com/assets/coco2017val.zip', 'tmp.zip')
+```
+and then unzip the file using:
+```bash
+unzip -q tmp.zip -d ../datasets && rm tmp.zip
+```
+
+Once done, we will then run the following  command to validate our model.
+
+```bash
+python val.py --weights yolov5x.pt --data coco.yaml --img 640 --iou 0.65 --half
+```
+Below is an example of our output:
+
+![evaluation](/engineering-education/building-an-object-detection-model-with-yolov5-and-pytorch/evaluation.jpg)
+
+The model's evaluation results will be saved to directory `runs/val/exp`.
+
+
+
+### Conclusion
+
+To wrap up, we have looked at what object detection is, its applications and implementation. We have also tackled the YOLO object detection algorithm (YOLOv5 particularly) which we have used to perform our own object detection from, setting up our environment to being detecting images and videos.
+
```

Comment:
Do we have the full code in the GitHub URL that you'd specified earlier?

Add colab notebook or notebook with all the code snippets executed.